service/dap/server.go

```diff
@@ -721,18 +729,53 @@ func (s *Server) onScopesRequest(request *dap.ScopesRequest) {
 	// Retrieve local variables
 	locals, err := s.debugger.LocalVariables(scope, cfg)
 	if err != nil {
-		s.sendErrorResponse(request.Request, UnableToListLocals, "Unable to list local vars", err.Error())
+		s.sendErrorResponse(request.Request, UnableToListLocals, "Unable to list locals", err.Error())
 		return
 	}
 	locScope := api.Variable{Name: "Locals", Children: locals}
 
 	// TODO(polina): Annotate shadowed variables
-	// TODO(polina): Retrieve global variables
 
 	scopeArgs := dap.Scope{Name: argScope.Name, VariablesReference: s.variableHandles.create(argScope)}
 	scopeLocals := dap.Scope{Name: locScope.Name, VariablesReference: s.variableHandles.create(locScope)}
 	scopes := []dap.Scope{scopeArgs, scopeLocals}
 
+	if s.args.loadGlobalVariables {
+		state, err := s.debugger.State( /*nowait*/ true)
+		if err != nil {
+			s.sendErrorResponse(request.Request, UnableToListGlobals, "Unable to list globals", err.Error())
+			return
+		}
+
+		// List global variables only for the current package only
+		currPkg := "??"
+		var globals []api.Variable
+		if state.CurrentThread != nil && state.CurrentThread.File != "" {
+			currDir := path.Dir(state.CurrentThread.File)
+			pkgInfo := s.debugger.ListPackagesBuildInfo( /*includeFiles*/ false)
```

Comment:
@hyangah
This is not how vscode-go gets the current package name, but I think the final product is the same. Please let me know if you think I missed something there.
https://github.com/golang/vscode-go/blob/c7c4188bb9bf048aeef803995ac94c582f435faa/src/debugAdapter/goDebug.ts#L1370

Reply:
@quoctruong

Reply:
I think the current DA behavior is ok (TBH, I often found this global variable section less useful during my debugging). I'd rather encourage users to set up watch list for the global variables they want to monitor instead of trying to prefetch all the global variables.

Reply:
Good point about the watch variables.
The current behavior makes sense to me, and this PR is aimed at reproducing it (expect I named the config flag loadGlobalVariables, not showGlobalVariables, since it seemed like "show" is very GUI-specific and delve can work with CLI, or do you think I should rename it back to show?). However, I am not using "go list" to get the package name like the current DA does, but instead use delve's equivalent. And since the mapping is done on delve's side, I don't think I need to do anything specific for the remote case. I just wanted to get another pair of eyes to verify my understanding that this alternative approach would yield the same results.